src/sentry/demo/middleware.py

```diff
@@ -1,33 +1,53 @@
 from django.conf import settings
 from django.core.urlresolvers import reverse
-from django.http import JsonResponse
+from django.http import JsonResponse, HttpResponseRedirect
 
 from sentry.models import OrganizationMember
 from sentry.utils import auth
 
 prompt_route = reverse("sentry-api-0-prompts-activity")
 org_creation_route = reverse("sentry-api-0-organizations")
+login_route = reverse("sentry-login")
+
+login_redirect_route = "https://sentry.io/welcome/"
 
 
 class DemoMiddleware:
     def process_view(self, request, view_func, view_args, view_kwargs):
         if not settings.DEMO_MODE:
             raise Exception("Demo mode misconfigured")
 
+        path = request.path
+        method = request.method
+
         # always return dismissed if we are in demo mode
-        if request.path == prompt_route:
+        if path == prompt_route:
             return JsonResponse({"data": {"dismissed_ts": 1}}, status=200)
 
         # disable org creation
-        if request.path == org_creation_route and request.method == "POST":
+        if path == org_creation_route and method == "POST":
             return JsonResponse(
                 {"detail": "Organization creation disabled in demo mode"}, status=400
             )
 
-        # only handling org views
+        # backdoor to allow logins
+        allow_login = request.GET.get("allow_login") != "1"
```

Comment:
Is this backwards? Seems weird that `allow_login != "1"` maps to the inverse.

Reply:
lol yea you're right

Reply:
@wedamija the logic was actually right but the variable name didn't make sense. Updated.